packages/iTwinUI-react/src/core/utils/components/Resizer.tsx

```diff
@@ -0,0 +1,298 @@
+          if (newWidth >= minWidth) {
+            width = elementRef.current.style.width = `${newWidth}px`;
+            translateX = initialTranslateX - (initialPointerX - clientX);
+          }
+          elementRef.current.style.transform = `translate(${translateX}px, ${translateY}px)`;
+          break;
+        }
+        case 'top': {
+          const newHeight = initialHeight + (initialPointerY - clientY);
+          if (newHeight < minHeight) {
+            break;
+          }
+          height = elementRef.current.style.height = `${newHeight}px`;
+          translateY = initialTranslateY - (initialPointerY - clientY);
+          elementRef.current.style.transform = `translate(${translateX}px, ${translateY}px)`;
+          break;
+        }
+        case 'top-right': {
+          const newHeight = initialHeight + (initialPointerY - clientY);
+          if (newHeight >= minHeight) {
+            height = elementRef.current.style.height = `${newHeight}px`;
+            translateY = initialTranslateY - (initialPointerY - clientY);
+          }
+          width = elementRef.current.style.width = `${
+            initialWidth - (initialPointerX - clientX)
+          }px`;
+          elementRef.current.style.transform = `translate(${translateX}px, ${translateY}px)`;
+          break;
+        }
+        case 'right': {
+          width = elementRef.current.style.width = `${
+            initialWidth - (initialPointerX - clientX)
+          }px`;
+          break;
+        }
+        case 'bottom-right': {
+          width = elementRef.current.style.width = `${
+            initialWidth - (initialPointerX - clientX)
+          }px`;
+          height = elementRef.current.style.height = `${
+            initialHeight - (initialPointerY - clientY)
+          }px`;
+          break;
+        }
+        case 'bottom': {
+          height = elementRef.current.style.height = `${
+            initialHeight - (initialPointerY - clientY)
+          }px`;
+          break;
+        }
+        case 'bottom-left': {
+          const newWidth = initialWidth + (initialPointerX - clientX);
+          if (newWidth >= minWidth) {
+            width = elementRef.current.style.width = `${newWidth}px`;
+            translateX = initialTranslateX - (initialPointerX - clientX);
+          }
+          height = elementRef.current.style.height = `${
+            initialHeight - (initialPointerY - clientY)
+          }px`;
+          elementRef.current.style.transform = `translate(${translateX}px, ${translateY}px)`;
+          break;
+        }
+        case 'left': {
+          const newWidth = initialWidth + (initialPointerX - clientX);
+          if (newWidth < minWidth) {
+            break;
+          }
+          width = elementRef.current.style.width = `${newWidth}px`;
+          translateX = initialTranslateX - (initialPointerX - clientX);
+          elementRef.current.style.transform = `translate(${translateX}px, ${translateY}px)`;
```

Comment:
oh this one is also a huge side effect, i think this should be handled in the dialog for sure. Do we even need to set it here if we are returning it in `setStyle`?

Reply:
I would disagree with this one cause the purpose of this component is to handle resizing and so it does that, also calling `setStyle` on each pointer move would be causing a lot of rerenders.

Reply:
But this relies on knowing that the thing that is being resized _always_ needs a transform applied. Rerenders are not always necessary because you can call this same line `style.transform = ...` inside Dialog without putting it in state.

And again, why are we returning it in setStyle currently if it will cause extra rerenders?

Reply:
`Resizer` code relies on `transform` cause its value is taken on resize start, so `Resizer` only works with `transform`. `setStyle` is needed to preserve resize state when `Dialog` is being closed by setting `isOpen={false}` and reopening again, the same as `useDragAndDrop`.